src/math/welch.js

```diff
@@ -15,14 +57,16 @@ import { nextpow2 } from './nextpow2';
  * @param {number} [options.overlapLength=null] - Amount of overlap between segments in samples. Defaults to floor(segmentLength / 2).
  * @param {string|number[]} [options.window='hann'] - Window function to apply, either 'hann', 'rectangular', or an array for a custom window. Default is 'hann'.
  * @param {number} [options.fftSize=Math.pow(2, bci.nextpow2(signal.length))] - Size of the fft to be used. Should be a power of 2.
+ * @param {string} [options.detrend='constant'] - Detrend to apply.
  * @returns {object} PSD object with keys {estimates: PSD estimates in units of X^2/Hz, frequencies: corresponding frequencies in Hz}
  */
 export function welch(signal, sample_rate, options) {
-    let { segmentLength, overlapLength, fftSize, window, verbose } = Object.assign({
+    let { segmentLength, overlapLength, fftSize, window, detrend, verbose } = Object.assign({
         segmentLength: 256,
         overlapLength: null,
         window: 'hann',
         fftSize: null,
+        detrend: 'constant',
```

Comment:
This is great, thank you so much! Let's make the default `none` for now, since any change in default behavior would require bumping the major version. Wouldn't want to break any users that already call this method and weren't expecting a detrend. We can make `constant` the default on the next major version bump

For now we can update the README.md / the Welch's method docstring to call this out and let users know to manually pass `detrend: 'constant'` if they want to match Scipy behavior